the-collab-lab/tcl-71-smart-shopping-list
# Guide user to add their first tiem

## Description

On the List Page, the user sees now a different interface displayed, depending of their experience with the application:
-> After creating their very first list, the user will see a message, a gif showing how to add an item, and a redirection button to the manage-list.
-> When a user creates a new list (and already has at least one), they will see a nice message and a redirection button to the manage-list.
-> When a user already has items in its list, it will display the items and the search component.

On Sign Out, the local storage is now being emptied.

## Related Issue

Closes [#8]

## Acceptance Criteria

 [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

Enhancement & bug fix

## Updates

### Before
![Screenshot 2024-02-27 154157](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/114222588/985c0b21-e2f8-4e88-b68b-86ae8f3522a3)


### After

### when a user is creating their first list
![Screenshot 2024-02-27 154542](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/114222588/ea5105e3-e262-4576-852c-0133d90830db)

### when a user has created a new list (and already has at least one list)
![Screenshot 2024-02-27 154138](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/114222588/6fae30de-c01b-4267-bfd3-bdd2d5ad2dd9)


### when a user has items present in their list
![Screenshot 2024-02-27 154157](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/114222588/985c0b21-e2f8-4e88-b68b-86ae8f3522a3)


## Testing Steps / QA Criteria

- git pull origin ce--ju--add-first-item
- npm start

List Page:
- Select a list and check whether the message displayed is adequate.
- Test with an existing user (once with empty list, once with items) and a newly created user.

Local Storage:
- On Logout, check your local storage, it should be empty.
